subsys/net/lib/sntp/sntp_pkt.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SNTP_PKT_H
+#define __SNTP_PKT_H
+
+#include <zephyr/types.h>
+
+struct sntp_pkt {
+	u8_t mode:3;		/* li, vn, and mode are big-endian */
```

Comment:
This was recently discussed on the mailing list - C standard doesn't guarantee order of bitfields within a machine field.

Reply:
It's fine if it's not part of a public API, and bit fields are really handy for network packet parsing. However I would encourage managing __BYTE_ORDER__ macro already here.

Reply:
You could have BIG/LITTLE endian check here although I am not sure how safe that is at the end.

Reply:
The problem occurs with public API, non-packed bitfields (though I am not entirely sure __packed garanties the order either), linking 2 objects using it but both compiled with 2 different compilers: you have no garanty that both objects use the same ordering.